scripts/jest.sh

```diff
@@ -7,6 +7,7 @@ NODE_ENV=test
 
 node \
   --expose-gc \
+  --no-compilation-cache \
```

Comment:
Once all tests were passing, I pushed up the changes and [observed](https://app.circleci.com/pipelines/github/artsy/force/47202/workflows/2242736e-5315-414d-a36c-365e1ebcb793/jobs/354979/parallel-runs/0?filterBy=FAILED) what appeared to be incremental heap usage increase during test runs. I quickly came across [this open issue](https://github.com/facebook/jest/issues/14042), very similar to what I observed in CI. Note that the open issue is also running node v18.15.0.

Upon further investigation, it appears that this issue affects [node v16.11.0+](https://github.com/facebook/jest/issues/11956) and may be node v8 engine problem.

Upon adding `--no-compilation-cache` flag the heap issue goes away. The test suite does seem a bit slower, according to some comments, up to 30%.

I wonder if this is a blocker for the upgrade and wanted to ask for input on this from others.

Reply:
Trying out the [workerIdleMemoryLimit](https://jestjs.io/docs/configuration/#workeridlememorylimit-numberstring) approach [here](https://github.com/artsy/force/compare/ovasdi/PLATFORM-5064/runtime-upgrade-jest-upgrade#diff-1e058ca1442e46581b13571fb8d261f9e1f5657e26c96634d4c1072f0f0347f1R30) (following jest upgrade). [Latest CI run](https://app.circleci.com/pipelines/github/artsy/force/47305/workflows/ecccbd16-90b8-469a-8b86-1a13ce31852a/jobs/355775).

Working through test failures introduced following the Jest upgrade.

Reply:
Resolved by updating jest and using a memory limit. See https://github.com/artsy/force/pull/12243/commits/e5291e22ed11650d87db16ea941553a238550941#r1171361094